clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/caching/Keys.groovy

```diff
@@ -77,37 +86,88 @@ class Keys implements KeyParser {
-        result << [id: parts[2], account: parts[3], region: parts[4], provider: parts[5]]
-        break
-      default:
-        return null
-        break
+    def result
+    result = [provider: parts[0], type: parts[1]]
+
+    if (parts[2] == CachingSchema.V2.toString() || parts[2] == CachingSchema.V2) { // parsing for split caching
+      switch (result.type) {
+        case Namespace.IMAGES.ns:
+          result << [account: parts[3], region: parts[4], imageId: parts[5]]
+          break
+        case Namespace.SERVER_GROUPS.ns:
+          def names = Names.parseName(parts[6])
+          result << [application: names.app.toLowerCase(), cluster: parts[3], account: parts[4], region: parts[5], serverGroup: parts[6], stack: names.stack, detail: names.detail, sequence: names.sequence?.toString()]
+          break
+        case Namespace.INSTANCES.ns:
+          result << [id: parts[3], region: parts[4], instanceId: parts[5]]
+          break
+        case Namespace.CLUSTERS.ns:
+          def names = Names.parseName(parts[5])
+          result << [application: parts[3].toLowerCase(), account: parts[4], cluster: parts[5], stack: names.stack, detail: names.detail]
+          break
+        case Namespace.APPLICATIONS.ns:
+          result << [application: parts[3].toLowerCase()]
+          break
+        case Namespace.HEALTH.ns:
+          result << [id: parts[2], account: parts[2], provider: parts[3]]
+        default:
+          return null
+          break
+      }
+    } else { // original parsing
+      switch (result.type) {
+        case Namespace.IMAGES.ns:
+          result << [account: parts[2], region: parts[3], imageId: parts[4]]
+          break
+        case Namespace.SERVER_GROUPS.ns:
+          def names = Names.parseName(parts[5])
+          result << [application: names.app.toLowerCase(), cluster: parts[2], account: parts[3], region: parts[4], serverGroup: parts[5], stack: names.stack, detail: names.detail, sequence: names.sequence?.toString()]
+          break
+        case Namespace.INSTANCES.ns:
+            result << [id: parts[2], region: parts[3], instanceId: parts[5]]
+          break
+        case Namespace.CLUSTERS.ns:
+          def names = Names.parseName(parts[4])
+          result << [application: parts[2].toLowerCase(), account: parts[3], cluster: parts[4], stack: names.stack, detail: names.detail]
+          break
+        case Namespace.APPLICATIONS.ns:
+          result << [application: parts[2].toLowerCase()]
+          break
+        case Namespace.HEALTH.ns:
+          result << [id: parts[2], account: parts[3], region: parts[4], provider: parts[5]]
+          break
+        default:
+          return null
+          break
+      }
     }
 
-    result
+    return result
+  }
+
+  // v2 keys for split provider
+  static String getImageV2Key(String imageId, String accountName, String region) {
+    "${TitusCloudProvider.ID}:${Namespace.IMAGES}:${CachingSchema.V2}:${accountName}:${region}:${imageId}"
```

Comment:
Adding a "V2" into the key to that we can parse correctly. Needs to be in the 3rd spot because search assumes first two keys are `cloudProvider:namespace`